tools/clang-tidy.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Define the find command to exclude certain directories and find relevant files
+
+# This command will exclude the following directories: same as format.sh but removing the opentelemetry-shim as well.
+# - third_party
+# - tools
+# - .git
+# - _deps
+# - build
+# - out
+# - .vs
+# - opentelemetry_logo.png
+# - TraceLoggingDynamic.h:
+# - opentelemetry-shim
+
+FIND="find . -name third_party -prune -o -name opentracing-shim -prune -o -name tools -prune -o -name .git -prune -o -name _deps -prune -o -name build -prune -o -name out -prune -o -name .vs -prune -o -name opentelemetry_logo.png -prune -o -name TraceLoggingDynamic.h -prune -o -name '*.h' -o -name '*.cc' -print"
```

Comment:
So, this script:

* finds all the code, whether included in the CMake build or not
* uses CMake to generate compile_commands.json
* Iterate on each file seen in the repository, to run clang-tidy on it.

This does not work well with generated code, code that is platform dependent, etc.

Please investigate this instead:

https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_CLANG_TIDY.html

The driver here should be CMake, which invokes clang-tidy on each file that are part of the build (WITH_OTLP_HTTP, etc)

Reply:
Thanks for the suggestion.  Added this flag:

 `-DCMAKE_CXX_CLANG_TIDY="clang-tidy-18"`
 
 This may be helpful if we wanted to do with each PR changes: [clang tidy diff](https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py)

Reply:
Checked locally that:

```
-DCMAKE_CXX_CLANG_TIDY="clang-tidy-unknown"
```

causes the build to fail when the binary is not found, so this validates that clang-tidy-18 effectively exists on the github runner, and is correct.